### PR TITLE
Fixed admission controller error handling for seed selection

### PR DESCRIPTION
**Pull request by afritzler**
Co-authored-by: Uwe Krueger <uwe.krueger@sap.com>

**What this PR does / why we need it**:
Provide a better message when the shoot network is not disjoint with the seed.

### CODE REVIEW

File: plugin/pkg/shoot/seedmanager/admission.go [hunk 1]
```diff
@@ -192,10 +209,6 @@ func determineSeed(shoot *garden.Shoot, seedLister gardenlisters.SeedLister, sho
 		}
 	}
 
-	if bestCandidate == nil {
```
Comment (rfranzke) [inline plugin/pkg/shoot/seedmanager/admission.go:195]: Why is this removed? I guess we need to keep it (to not `return nil, nil`), and maybe just improve the error message?

Comment (afritzler) [reply plugin/pkg/shoot/seedmanager/admission.go:195]: We can add it at the end to be on the safe side. But a `nil, nil` return can never happen if we have at least 1 `candidate`. Otherwise we exit with an error before that.

Comment (afritzler) [reply plugin/pkg/shoot/seedmanager/admission.go:195]: We added this PR because we ran into an issue where we had colliding CIDRs and the error message of the admission was extremely misleading.

**Comment by afritzler** [comment]
Ok, I've added back the `bestCandidate` check.

File: plugin/pkg/shoot/seedmanager/admission.go (hunk 1 above)
Comment (rfranzke) [reply plugin/pkg/shoot/seedmanager/admission.go:195]: > We can add it at the end to be on the safe side. But a `nil, nil` return can never happen if we have at least 1 `candidate`. Otherwise we exit with an error before that.

I checked again, you're right, can never happen. So let's go back what you did before and completely remove that `if`-block. Thank you!

> We added this PR because we ran into an issue where we had colliding CIDRs and the error message of the admission was extremely misleading.

Sure, agreed! That's definitely an improvement - thank you!

File: plugin/pkg/shoot/seedmanager/admission.go [hunk 2]
```diff
@@ -174,11 +174,28 @@ func determineSeed(shoot *garden.Shoot, seedLister gardenlisters.SeedLister, sho
...
+		if seed.DeletionTimestamp == nil && seed.Spec.Cloud.Profile == shoot.Spec.Cloud.Profile && seed.Spec.Cloud.Region == shoot.Spec.Cloud.Region && seed.Spec.Visible != nil && *seed.Spec.Visible && verifySeedAvailability(seed) {
 			candidates = append(candidates, seed)
 		}
 	}
 
+	if candidates == nil {
+		return nil, errors.New("no adequate seed cluster found for this cloud profile and region")
+	}
+
+	old := candidates
```
Comment (rfranzke) [inline plugin/pkg/shoot/seedmanager/admission.go:186]: Should we use `old := append(candidates[:0:0], candidates...)` here (as stated [here](https://github.com/go101/go101/wiki/How-to-perfectly-clone-a-slice%3F))?
@adracus opinion?

File: plugin/pkg/shoot/seedmanager/admission.go (hunk 1 above)
Comment (afritzler) [reply plugin/pkg/shoot/seedmanager/admission.go:195]: Alright. Will revert the stuff.

File: plugin/pkg/shoot/seedmanager/admission.go (hunk 2 above)
Comment (afritzler) [reply plugin/pkg/shoot/seedmanager/admission.go:186]: Good point!

Comment (afritzler) [reply plugin/pkg/shoot/seedmanager/admission.go:186]: We actually don't copy the slice here. So no need to `append`.

Comment (rfranzke) [reply plugin/pkg/shoot/seedmanager/admission.go:186]: OK. Please squash the commits, then we can get it merged!

Comment (afritzler) [reply plugin/pkg/shoot/seedmanager/admission.go:186]: Done!

**Review by rfranzke — APPROVED** [review]
/lgtm
